Sources/BraveWallet/Crypto/Stores/AssetDetailStore.swift

```diff
@@ -93,63 +132,118 @@ class AssetDetailStore: ObservableObject {
     walletService.defaultBaseCurrency { [self] currencyCode in
       self.currencyCode = currencyCode
     }
+    
+    fetchSupportedBuyTokens()
   }
 
   private let percentFormatter = NumberFormatter().then {
     $0.numberStyle = .percent
     $0.maximumFractionDigits = 2
   }
   
+  private func fetchSupportedBuyTokens() {
```

Comment:
I think this function can probably be made a function that accepts a `NetworkInfo` model and returns the `[BraveWallet.OnRampProvider : [BraveWallet.BlockchainToken]]` dictionary. This way it can be re-used for nboth `AssetDetailType.blockchainToken` and `AssetDetailType.coinMarket` cases. Seems like we only have a `en-us` Sardine logic for `coinMarket` and not `blockchainToken` 🤔.